File: lib/data/non/lua/network.lua
non.network = {}
non.network.module = NON_MODULE:get("network")
  
function non.network.client()
  return non.network.module:client()
end
  
function non.network.server()
  return non.network.module:server()
end
  
function non.network.buffer(data)
  if data == nil then
    return non.network.module:buffer()
  else
    return non.network.module:buffer(data)
  end
end

File: lib/data/non/ruby/network.rb
class Network
  java_import 'non.Buffer'
  
  def initialize
    @module = NON::JModule.get("network")
  end
  
  def client
    @module.client()
  end
  
  def server
    @module.server()
  end
  
  def buffer(data = nil)
    data == nil ? Buffer.new : Buffer.new(data)
  end
end